beacon-chain/core/blocks/payload_test.go

```diff
@@ -409,6 +413,106 @@ func Test_ExecutionEnabled(t *testing.T) {
 	}
 }
 
+func Test_ValidatePayloadWhenMergeCompletes(t *testing.T) {
+	tests := []struct {
+		name    string
+		payload *ethpb.ExecutionPayload
+		header  *ethpb.ExecutionPayloadHeader
+		err     error
+	}{
+		{
+			name:    "merge incomplete",
+			payload: emptyPayload(),
+			header:  emptyPayloadHeader(),
+			err:     nil,
+		},
+		{
+			name:    "validate passes",
+			payload: emptyPayload(),
+			header: func() *ethpb.ExecutionPayloadHeader {
+				h := emptyPayloadHeader()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
+				return h
+			}(),
+			err: nil,
+		},
+		{
+			name: "incorrect blockhash",
+			payload: func() *ethpb.ExecutionPayload {
+				h := emptyPayload()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
+				return h
+			}(),
+			header: func() *ethpb.ExecutionPayloadHeader {
+				h := emptyPayloadHeader()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
+				return h
```

Comment:
1. You should set the payload's `BlockHash`
2. These two should not be equal. Currently both are `a`